trusted-firmware-m/cmake/Common/CompilerGNUARMCommon.cmake

```diff
@@ -146,6 +146,7 @@ function(compiler_set_cmse_output TARGET FILE_PATH)
 	set_property(TARGET ${TARGET} APPEND_STRING PROPERTY LINK_FLAGS " -Wl,--cmse-implib,--out-implib=${FILE_PATH}")
 	#Tell cmake cmse output is a generated object file.
 	SET_SOURCE_FILES_PROPERTIES("${FILE_PATH}" PROPERTIES EXTERNAL_OBJECT true GENERATED true)
+	add_custom_command(TARGET ${TARGET} POST_BUILD COMMAND ${CMAKE_COMMAND} -E echo "" BYPRODUCTS ${FILE_PATH})
```

Comment:
is this just debug?

Reply:
No, it's because the `${FILE_PATH}` which is the `s_veneers.o` file is created as part of the final linking.
But the linking, which happens because of the `add_executable()` does not support the `BYPRODUCTS` keyword.

This is done manually by controlling the properties on the files, and using `add_dependencies`, which works well for `make` but `ninja` is a bit more strict, and want to know a specific target that can create the file.

Therefore `add_custom_command()` is used, just to make `ninja` happy.
It doesn't need to do anything, but it provides the needed relation between the file and the target.